app/views/unattended/provisioning_templates/registration/linux_registration_default.erb

```diff
@@ -20,6 +20,19 @@ set -e
 <%= snippet 'puppet_setup' %>
 <% end -%>
 
+<% if host_param_true?('host_registration_insights') -%>
+echo '#'
+echo '# Installing Insights client'
+echo '#'
+
+yum install insights-client
+insights-client --register
+<% end -%>
+
+<% if host_param_true?('host_registration_rex') -%>
```

Comment:
I'd drop this parameter since all others also call it unconditionally:
https://github.com/theforeman/foreman/blob/fc2387648abc558d8f3a8d9f4d914530f5823587/app/views/unattended/provisioning_templates/finish/kickstart_default_finish.erb#L58-L60

If we want a parameter to control that, it should behave the same for all methods (provisioning, registration, whatever). That is out of scope for this PR so I'd leave it out.